### PR TITLE
Seasonal booking datamodel

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Updates the seasonal booking data model and related APIs to be more aligned with the current process
- Admin updated with new seasonal booking stuff & translations
- Add endpoints or locking/rejecting reservation unit options
- Refactors permissions
- Adds terms of use to application rounds

TODO: 
- Fixing bugs if frontend notices any more
- Tracking down "performance killers" for new SQL optimizer

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

### Noted for PR reviewers
- Automated tests should pass
- Run migrations, test data generation, and check that things "still work"

### Notes for frontend:
- Complain if the new test data generation needs improvements
- Backend endpoints for some of the seasonal booking stuff is now optimized for database queries, but since I didn't make the change everywhere it's not as good as it should be.

### Notes for manual testers:
- Manual testing should also check generally that "all pages" still load. New libraries changed how automatic field generation works, so there might be subtle changes, which we didn't catch with previous testing. A thorough check is not required.
- Should check the usage of the reservation priority field together with the frontend.
- Should check that seasonal booking statuses work as expected. No docs on this yet, so ask me to explain when ready.
- Should take note of the permission-related changes, but conduct a thorough permission review at a later point.
- Should double-check that the application round reservation period now actually blocks reservations for its reservation period. Seems based on previous code that this was done incorrectly.

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-3133](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3133)
- [TILA-3181](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3181)
- [TILA-2264](https://helsinkisolutionoffice.atlassian.net/browse/TILA-2264)

[TILA-3133]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/test_models/test_application.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    assert application.status == ApplicationStatusChoice.DRAFT
+    assert Application.objects.filter(L(status=ApplicationStatusChoice.DRAFT)).exists()
+
+    # Application round has moved to allocation, but application was not sent -> application is EXPIRED
+    application_round.application_period_end = now - datetime.timedelta(days=1)
+    application_round.save()
+    assert application.status == ApplicationStatusChoice.EXPIRED
+    assert Application.objects.filter(L(status=ApplicationStatusChoice.EXPIRED)).exists()
+
+    # Application was sent, but without any sections -> application is IN_ALLOCATION
```
Comment (vergama) [inline tests/test_models/test_application.py:43]: HANDLED

File: locale/sv/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -416,9 +410,88 @@ msgstr ""
...
+msgstr "Varausyksikkö vaihtoehto"
+
+#: applications/models/reservation_unit_option.py
+msgid "Reservation Unit Options"
+msgstr "Varausyksikkö vaihtoehdot"
+
+#: applications/models/reservation_unit_option.py
+msgid "Preferred order must be unique for each application section"
+msgstr "Halutun järjestyksen tule olla uniikki jokaisella hakemuksen osalla"
+
```
Comment (vergama) [inline locale/sv/LC_MESSAGES/django.po:483]: ...tule**e** olla...

File: locale/sv/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -416,9 +410,88 @@ msgstr ""
...
+"ja alle 24 tuntia."
+
 #: applications/models/organization.py
 msgid "is after current year"
-msgstr ""
+msgstr "on tämän vuoden jälkeen"
+
+#: applications/models/reservation_unit_option.py
+msgid "Reservation Unit Option"
+msgstr "Varausyksikkö vaihtoehto"
```
Comment (vergama) [inline locale/sv/LC_MESSAGES/django.po:474]: Varausyksikkö-vaihtoehto

File: locale/sv/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -416,9 +410,88 @@ msgstr ""
...
+
+#: applications/models/reservation_unit_option.py
+msgid "Reservation Unit Options"
+msgstr "Varausyksikkö vaihtoehdot"
```
Comment (vergama) [inline locale/sv/LC_MESSAGES/django.po:478]: Varausyksikkö-vaihtoehdot

Comment (matti-lamppu) [reply locale/sv/LC_MESSAGES/django.po]: Done.

File: locale/sv/LC_MESSAGES/django.po (hunk 3 above)
Comment (matti-lamppu) [reply locale/sv/LC_MESSAGES/django.po]: Done.

File: locale/sv/LC_MESSAGES/django.po (hunk 2 above)
Comment (matti-lamppu) [reply locale/sv/LC_MESSAGES/django.po:483]: Done.

File: tests/test_graphql_api/test_application_section/test_ordering.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+    section_3 = ApplicationSectionFactory.create_in_status_unallocated()
+    ReservationUnitOptionFactory.create(
+        application_section=section_3,
+        preferred_order=1,
+        **{f"reservation_unit__unit__name_{lang}": "A unit"},
+    )
+
+    # Section doesn't have reservation unit options at all
+    # -> preferred_unit_name is None -> ordered last
+    event_4 = ApplicationSectionFactory.create_in_status_unallocated()
```
[the remark this answers is not in the document]
Comment (matti-lamppu) [reply tests/test_graphql_api/test_application_section/test_ordering.py]: Done.

File: tests/test_models/test_application.py (hunk 1 above)
Comment (matti-lamppu) [reply tests/test_models/test_application.py]: Done.

File: api/graphql/types/allocated_time_slot/serializers.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+
+    def validate_statuses(
+        self,
+        section: ApplicationSection,
+        errors: dict[str, list[str]],
+    ) -> None:
+        section_status = section.status
+        application_status = section.application.status
+
+        if not section_status.can_approve:
```
Comment (vergama) [inline api/graphql/types/allocated_time_slot/serializers.py:97]: I guess section_status.can_approve could be renamed can_allocate

Comment (matti-lamppu) [reply api/graphql/types/allocated_time_slot/serializers.py]: Done.

**Review by ranta — APPROVED** [review]

File: locale/fi/LC_MESSAGES/django.po [hunk 7]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
+msgid "Minimum duration that should be allocated for this section."
+msgstr "Minimikesto allokoitaville vuoroille."
+
+#: applications/admin/forms/application_section.py
+msgid "Maximum duration that should be allocated for this section."
+msgstr "Maksimikesto allokoitaville vuoroille."
+
+#: applications/admin/forms/application_section.py
+msgid "First date on which reservations for this section are created."
+msgstr "Ensimmäinen päivä jolle varaukset tehdään allokoitien perusteella."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:534]: allokoitien > allokointien

File: locale/fi/LC_MESSAGES/django.po [hunk 8]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
+msgid "Purpose for this section."
+msgstr "Hakemuksen osan käyttötarkoitus."
+
+#: applications/admin/forms/application_section.py
+msgid "Suitable days of the week"
+msgstr "Sopivat viikonpäivät"
+
+#: applications/admin/forms/application_section.py
+msgid "Suitable days of the week for this section."
+msgstr "Hakemusen osalle sopivat viikonpäivät."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:558]: Hakemusen > Hakemuksen

File: locale/fi/LC_MESSAGES/django.po [hunk 9]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
+msgid "Email"
+msgstr "Sähköposti"
+
+#: applications/admin/forms/organization.py merchants/models.py
+msgid "Company code"
+msgstr "Y-tunnus"
+
+#: applications/admin/forms/organization.py
+msgid "Year established"
+msgstr "Perustemisvuosi"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:582]: Perustamis

File: locale/fi/LC_MESSAGES/django.po [hunk 10]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
+
+#: applications/admin/forms/organization.py
+msgid "Active members"
+msgstr "Aktiivisia jäseniä"
+
+#: applications/admin/forms/organization.py
+msgid "Core business area"
+msgstr "Ydin liiketoiminta-alue"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:590]: Ydinliiketoiminta-alue

File: locale/fi/LC_MESSAGES/django.po [hunk 11]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: common/admin/forms/banner_notification.py
 msgid "Level of the notification."
-msgstr ""
+msgstr "Ilmoituksen taso."
 
 #: common/admin/forms/banner_notification.py
 msgid "Target user interface of the notification."
-msgstr ""
+msgstr "Ilmoituksen kohde käyttöliittymä."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1006]: kohde > kohdistettu

File: locale/fi/LC_MESSAGES/django.po [hunk 12]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
-
-#: common/choices/banner_notification.py merchants/models.py
-msgid "Draft"
-msgstr ""
+msgstr "Käyttäjä"
 
 #: common/choices/banner_notification.py
 msgid "Scheduled"
-msgstr ""
+msgstr "Ajoitettu"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1053]: Ajastettu

File: locale/fi/LC_MESSAGES/django.po [hunk 13]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 "Email body content as HTML. Use curly brackets to indicate data specific "
 "fields e.g {{reservee_name}}."
 msgstr ""
+"Sähköpostin sisältö HTML-muodossa. Käytä aaltosulkeita osoittamaan dataan "
+"liittyviä kenttiä, esim {{reservee_name}}."
 
 #: merchants/admin.py
 msgid "Paytrail merchant ID"
-msgstr ""
+msgstr "Paytrail kauppiastunnus"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1114]: Paytrail-kauppiastunnus

File: locale/fi/LC_MESSAGES/django.po [hunk 14]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: merchants/admin.py
 msgid "The Paytrail Merchant ID should be a six-digit number."
-msgstr ""
+msgstr "Paytrail kauppiastunnuksen tulisi olla kuusinumeroinen."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1118]: Paytrail-kauppiastunnuksen

File: locale/fi/LC_MESSAGES/django.po [hunk 15]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: merchants/models.py
 msgid "Remote order ID"
-msgstr ""
+msgstr "Ulkoinen tilaustunniste"
 
 #: merchants/models.py
 msgid "eCommerce order ID"
-msgstr ""
+msgstr "eCommerce tilaustunniste"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1214]: eCommerce-tilaustunniste

File: locale/fi/LC_MESSAGES/django.po [hunk 16]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: merchants/models.py
 msgid "Payment ID"
-msgstr ""
+msgstr "Maksutunniste"
 
 #: merchants/models.py
 msgid "eCommerce payment ID"
-msgstr ""
+msgstr "eCommerce maksutunniste"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1222]: eCommerce-maksutunniste

File: locale/fi/LC_MESSAGES/django.po [hunk 17]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: merchants/models.py
 msgid "Checkout URL"
-msgstr ""
+msgstr "Kassan URL"
 
 #: merchants/models.py
 msgid "Receipt URL"
-msgstr ""
+msgstr "Kuittauksen URL"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1275]: Kuitin URL

File: locale/fi/LC_MESSAGES/django.po [hunk 18]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 msgid ""
 "OriginHaukiResources with this specific hash never have any opening hours."
 msgstr ""
+"OriginHaukiResources, joilla on tämä tietty tarkiste, eivät koskaan sisällä "
+"aukioloaikoja."
 
 #: opening_hours/admin.py
 msgid "ID of the resource in Hauki."
-msgstr ""
+msgstr "Resurssin tunniste Haukista."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1339]: Resurssin tunniste Aukiolosovelluksesta.

File: locale/fi/LC_MESSAGES/django.po [hunk 19]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 msgstr ""
+"Aukioloaikojen tarkiste. Käytetään määrittämään, ovatko aukioloajat "
+"muuttuneet."
 
 #: opening_hours/admin.py
 msgid ""
 "All opening hours have been fetched from Hauki up until this date. Opening "
 "hours are fetched until the last day of the month two years from now."
 msgstr ""
+"Kaikki aukioloajat on haettu Haukista tähän päivämäärään asti. Aukioloajat "
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1353]: ... Aukiolosovelluksesta ...

File: locale/fi/LC_MESSAGES/django.po [hunk 20]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 #: reservation_units/models/reservation_unit.py
 #: reservation_units/models/reservation_unit_image.py
 msgid "Type"
-msgstr ""
+msgstr "Tyyppi"
 
 #: reservation_units/models/reservation_unit.py
 msgid "Require introduction"
-msgstr ""
+msgstr "Vaatii perehdytys"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:1831]: Vaatii perehdytyksen

File: locale/fi/LC_MESSAGES/django.po [hunk 21]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing first name"
-msgstr ""
+msgstr "Laskuttajan etunimi"
 
 #: reservations/models/reservation.py
 msgid "Billing last name"
-msgstr ""
+msgstr "Lasuttajan sukunimi"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2165]: Laskuttajan

File: locale/fi/LC_MESSAGES/django.po [hunk 22]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing phone"
-msgstr ""
+msgstr "Lasuttajan puhelinnumero"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2170]: Laskuttajan

File: locale/fi/LC_MESSAGES/django.po [hunk 23]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing email"
-msgstr ""
+msgstr "Lasuttajan sähköposti"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2173]: Laskuttajan

File: locale/fi/LC_MESSAGES/django.po [hunk 24]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing address street"
-msgstr ""
+msgstr "Lasuttajan katuosoite"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2177]: Laskuttajan

File: locale/fi/LC_MESSAGES/django.po [hunk 25]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing address city"
-msgstr ""
+msgstr "Lasuttajan kaupunki"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2181]: Laskuttajan

File: locale/fi/LC_MESSAGES/django.po [hunk 26]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
-#: reservations/models/reservation.py
-#: reservations/models/reservation_statistic.py
-msgid "Home city"
-msgstr ""
-
-#: reservations/models/reservation.py
-#: reservations/models/reservation_statistic.py
-msgid "Age group"
-msgstr ""
+msgstr "Lasuttajan postinumero"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2185]: Laskuttajan

File: locale/fi/LC_MESSAGES/django.po [hunk 27]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 #: reservations/models/reservation_statistic.py
 msgid "Reservee is applying for a free-of-charge reservation"
-msgstr ""
+msgstr "Lasuttaja hakee alennuskelpoisuutta"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2190]: Laskuttaja

File: locale/fi/LC_MESSAGES/django.po [hunk 28]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
-
 #: reservations/models/reservation_metadata.py
 msgid "Field name"
-msgstr ""
+msgstr "Kentän nimi"
 
 #: reservations/models/reservation_metadata.py
 msgid "Reservation metadata field"
-msgstr ""
+msgstr "Varauslomakekenttä"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2284]: Varauslomakkeen kenttä

File: locale/fi/LC_MESSAGES/django.po [hunk 29]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_metadata.py
 msgid "Reservation metadata fields"
-msgstr ""
+msgstr "Varauslomakekentät"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2288]: Varauslomakkeen kentät

File: locale/fi/LC_MESSAGES/django.po [hunk 30]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_metadata.py
 msgid "Required fields"
-msgstr ""
+msgstr "Vaaditut kentät"
 
 #: reservations/models/reservation_metadata.py
 msgid "Reservation metadata sets"
-msgstr ""
+msgstr "Varauslomakekenttä setit"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2300]: Varauslomakkeen kenttäkokonaisuus

File: locale/fi/LC_MESSAGES/django.po [hunk 31]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
-
-#: reservation_units/models/reservation_unit_pricing.py
-msgid "Status"
-msgstr ""
+msgstr "Veroprosentti"
 
 #: reservations/admin.py
 msgid "Required fields must be a subset of supported fields"
-msgstr ""
+msgstr "Vaaditut kentät täytyy olla tuettujen kenttien alijoukko"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2079]: Vaadittujen kenttien ...

File: locale/fi/LC_MESSAGES/django.po [hunk 32]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
+msgstr "Syy hylkäämiselle"
 
 #: reservations/models/reservation.py
 msgid "Handling details for this reservation"
-msgstr ""
-
-#: reservations/models/reservation.py
-msgid "Working memo"
-msgstr ""
+msgstr "Käsittelytiedot tälle varaukselle"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2272]: Käsittelymuistio ...

File: locale/fi/LC_MESSAGES/django.po [hunk 33]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_statistic.py
 msgid "The reason text of the deny reason"
-msgstr ""
+msgstr "Hylkäämisen syyteksti"
 
 #: reservations/models/reservation_statistic.py
 msgid "TPREK id"
-msgstr ""
+msgstr "TPREK id"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2336]: Toimipisterekisterin tunniste

File: locale/fi/LC_MESSAGES/django.po [hunk 34]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: services/models.py
 msgid "Service type"
-msgstr ""
+msgstr "Palvelun tyyppi"
 
 #: spaces/admin.py
 msgid "Selected units for this unit group."
-msgstr ""
+msgstr "Valitut yksiköt tälle yksikköryhmälle."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2364]: Valitut toimipisteet tälle toimipisteryhmälle.

File: locale/fi/LC_MESSAGES/django.po [hunk 35]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: spaces/admin.py
 msgid "Previously selected number of units for this unit group."
-msgstr ""
+msgstr "Aiemmin valittu yksiköiden määrä tälle yksikköryhmälle."
 
 #: spaces/admin.py
 msgid "Name of the unit group in Finnish."
-msgstr ""
+msgstr "Yksikköryhmän nimi suomeksi."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2373]: Toimipisteryhmän

File: locale/fi/LC_MESSAGES/django.po [hunk 36]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: spaces/admin.py
 msgid "Name of the unit group in English."
-msgstr ""
+msgstr "Yksikköryhmän nimi englanniksi."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2377]: Toimipisteryhmän

File: locale/fi/LC_MESSAGES/django.po [hunk 37]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: spaces/admin.py
 msgid "Name of the unit group in Swedish."
-msgstr ""
+msgstr "Yksikköryhmän nimi ruotsiksi."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2381]: Toimipisteryhmän

File: locale/fi/LC_MESSAGES/django.po [hunk 38]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: spaces/admin.py
 msgid "Selected units for this service sector."
-msgstr ""
+msgstr "Valitut yksiköt tälle palvelualueelle."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2384]: Toimipisteryhmän

File: locale/fi/LC_MESSAGES/django.po [hunk 39]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: spaces/admin.py
 msgid "Previously selected number of units for this service sector."
-msgstr ""
+msgstr "Aiemmin valittu yksiköiden määrä tälle palvelualueelle."
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2388]: ... toimipisteiden ...

File: locale/fi/LC_MESSAGES/django.po [hunk 40]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
+
+#: applications/admin/allocated_time_slot.py applications/admin/application.py
+#: applications/admin/forms/application_section.py
+#: applications/models/application.py
+msgid "Application"
+msgstr "Hakemus"
+
+#: applications/admin/application.py
+msgid "Reset application allocations"
+msgstr "Resetoi hakemuksen allokoinnit"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:48]: Nollaa hakemuksen allokoinnit

File: locale/fi/LC_MESSAGES/django.po [hunk 41]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: applications/admin/application.py applications/admin/application_round.py
 msgid "Are you sure?"
-msgstr ""
+msgstr "Oletko varma?"
 
 #: applications/admin/application.py applications/admin/application_round.py
 msgid "Are you sure you want reset allocations?"
-msgstr ""
+msgstr "Oletko varma, että haluat resetoida allokoinnit?"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:56]: ... nollata allokoinnit?

File: locale/fi/LC_MESSAGES/django.po (hunk 27 above)
Comment (matti-lamppu) [reply locale/fi/LC_MESSAGES/django.po:2190]: Ehkä ennemmin "Varaaja"? (käännökset on suurimmaksi osaksi ChatGPT:n muuten)

File: locale/fi/LC_MESSAGES/django.po [hunk 42]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
-#: reservations/models/reservation.py
-#: reservations/models/reservation_statistic.py
-msgid "Home city"
-msgstr ""
-
-#: reservations/models/reservation.py
-#: reservations/models/reservation_statistic.py
-msgid "Age group"
-msgstr ""
+msgstr "Laskuttajan postinumero"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2185]: Varauksen maksajan postinumero

File: locale/fi/LC_MESSAGES/django.po [hunk 43]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing address street"
-msgstr ""
+msgstr "Laskuttajan katuosoite"
 
 #: reservations/models/reservation.py
 msgid "Billing address city"
-msgstr ""
+msgstr "Laskuttajan kaupunki"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2181]: Varauksen maksajan postitoimipaikka

File: locale/fi/LC_MESSAGES/django.po [hunk 44]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing email"
-msgstr ""
+msgstr "Laskuttajan sähköposti"
 
 #: reservations/models/reservation.py
 msgid "Billing address street"
-msgstr ""
+msgstr "Laskuttajan katuosoite"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2178]: Varauksen maksajan katuosoite

File: locale/fi/LC_MESSAGES/django.po [hunk 45]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing phone"
-msgstr ""
+msgstr "Laskuttajan puhelinnumero"
 
 #: reservations/models/reservation.py
 msgid "Billing email"
-msgstr ""
+msgstr "Laskuttajan sähköposti"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2173]: Varauksen maksajan sähköposti

File: locale/fi/LC_MESSAGES/django.po [hunk 46]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing last name"
-msgstr ""
+msgstr "Laskuttajan sukunimi"
 
 #: reservations/models/reservation.py
 msgid "Billing phone"
-msgstr ""
+msgstr "Laskuttajan puhelinnumero"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2169]: Varauksen maksajan puhelinnumero

File: locale/fi/LC_MESSAGES/django.po [hunk 47]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation.py
 msgid "Billing first name"
-msgstr ""
+msgstr "Laskuttajan etunimi"
 
 #: reservations/models/reservation.py
 msgid "Billing last name"
-msgstr ""
+msgstr "Laskuttajan sukunimi"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2165]: Varauksen maksajan sukunimi

File: locale/fi/LC_MESSAGES/django.po [hunk 48]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 #: reservations/models/reservation.py
 #: reservations/models/reservation_statistic.py
 msgid "Preferred language of reservee"
-msgstr ""
+msgstr "Varaajan kieli"
 
 #: reservations/models/reservation.py
 msgid "Billing first name"
-msgstr ""
+msgstr "Laskuttajan etunimi"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2161]: Varauksen maksajan etunimi

File: locale/fi/LC_MESSAGES/django.po [hunk 49]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_statistic.py
 msgid "Home city name"
-msgstr ""
+msgstr "Kotikaupungin nimi"
 
 #: reservations/models/reservation_statistic.py
 msgid "Home city municipality code"
-msgstr ""
+msgstr "Kotikaupungin kunnan koodi"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2312]: Kotikunnan kuntanumero

File: locale/fi/LC_MESSAGES/django.po [hunk 50]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_statistic.py
 msgid "Statistics updated at"
-msgstr ""
+msgstr "Tilastot päivitetty"
 
 #: reservations/models/reservation_statistic.py
 msgid "Home city name"
-msgstr ""
+msgstr "Kotikaupungin nimi"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2308]: Kotikunnan nimi

File: locale/fi/LC_MESSAGES/django.po [hunk 51]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_statistic.py
 msgid "Reservation duration in minutes"
-msgstr ""
+msgstr "Varauksen kesto minuutteina"
 
 #: reservations/models/reservation_statistic.py
 msgid "The reason text of the cancel reason"
-msgstr ""
+msgstr "Perumisen syyteksti"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2328]: ... syyn kuvaus

File: locale/fi/LC_MESSAGES/django.po [hunk 52]
```diff
@@ -31,1905 +21,2446 @@ msgstr "Käyttäjällä on avoimia maksuja."
...
 
 #: reservations/models/reservation_statistic.py
 msgid "The reason text of the deny reason"
-msgstr ""
+msgstr "Hylkäämisen syyteksti"
```
Comment (vergama) [inline locale/fi/LC_MESSAGES/django.po:2332]: ... syyn kuvaus

**Review by vergama — CHANGES_REQUESTED** [review]
All typo fixes and some other translation changes now commented

**Comment by matti-lamppu** [comment]
@vergama Translations updated

**Review by vergama — APPROVED** [review]

**Pull request merged**